### ISSUE TITLE
Question: Tools for 2D rigging?

### ISSUE THREAD

**Issue by Johferson** [FunkinCrew/Funkin#2477]
[weed]: <> (This isn't a place for AMA type questions, if you want to ask any of the devs something, reach out to them on twitter prob )
[weed]: <> (any biz bullshit can go to cameron.taylor.ninja@gmail.com)

## What is your question?
Hi Funkin' Crew Inc. I just wanted to ask which tools did you guys use for rigging some 2D sprites like in Pico's game over, Blazin', BF in freeplay, and basically everything else that was added after Week 7. Maybe modders can use some of those tools to make future mods efficient and less storage-intensive. Thank you!

**Comment by biomseed** [comment]
It's not a "rigging" tool. It's a way to export animations called a "texture atlas" that is very memory efficient and space efficient. You can do it right now in adobe animate.

**Comment by Johferson** [comment]
> It's not a "rigging" tool. It's a way to export animations called a "texture atlas" that is very memory efficient and space efficient. You can do it right now in adobe animate.

Fool! Texture Atlases are the spritesheets itself for the legacy stuff (and spritesheets in general). I am talking about how they animated recent stuff using separate assets as a 2D skeletal animation to make it seem like it's one asset instead of individual assets.

An example of what I am saying is Plants vs Zombies. They are all animated skeletally in 2D using individual assets instead of having all of them be using Texture Atlases/spritesheets

**Comment by biomseed** [comment]
> > It's not a "rigging" tool. It's a way to export animations called a "texture atlas" that is very memory efficient and space efficient. You can do it right now in adobe animate.
> 
> Texture Atlases are the spritesheets itself for the legacy stuff (and spritesheets in general). I am talking about how they animated recent stuff using separate assets as a 2D skeletal animation to make it seem like it's one asset instead of individual assets.
> 
> An example of what I am saying is Plants vs Zombies. They are all animated skeletally in 2D using individual assets instead of having all of them be using Texture Atlases/spritesheets, pretty similar to the recent FNF stuff added

That's not what texture atlases are. Texture atlases is this stuff:
![spritemap1](https://github.com/FunkinCrew/Funkin/assets/73907153/8d421577-92c7-441d-b02d-8d8a5177626c)
I know the spritesheets are technically a texture atlas, but the option to export a "texture atlas" and the option to export a "sprite sheet" are two different things in adobe animate

**Comment by Johferson** [comment]
> > > It's not a "rigging" tool. It's a way to export animations called a "texture atlas" that is very memory efficient and space efficient. You can do it right now in adobe animate.
> > 
> > 
> > Texture Atlases are the spritesheets itself for the legacy stuff (and spritesheets in general). I am talking about how they animated recent stuff using separate assets as a 2D skeletal animation to make it seem like it's one asset instead of individual assets.
> > An example of what I am saying is Plants vs Zombies. They are all animated skeletally in 2D using individual assets instead of having all of them be using Texture Atlases/spritesheets, pretty similar to the recent FNF stuff added
> 
> That's not what texture atlases are. Texture atlases is this stuff: ![spritemap1](https://private-user-images.githubusercontent.com/73907153/329898499-8d421577-92c7-441d-b02d-8d8a5177626c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU1OTU2MTksIm5iZiI6MTcxNTU5NTMxOSwicGF0aCI6Ii83MzkwNzE1My8zMjk4OTg0OTktOGQ0MjE1NzctOTJjNy00NDFkLWIwMmQtOGQ4YTUxNzc2MjZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTEzVDEwMTUxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2ZmUzMmUwZjY1ODAwMTdiZGE3NDUwZjUwMDBlMjZhZWE4MjJhNGU3ZGIwZjg2ODgyOTg2MzUzNTE2YTQ1NDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.sDAp42dscmhgU-NnvVwrp0HwKcqvYuNJEpQAC0pJ0Us) I know the legacy spritesheets are technically a texture atlas, but the option to export a "texture atlas" and the option to export a "sprite sheet" are two different things in adobe animate

Yeah I just found out now. The only question now is if there would be a way to add in objects that are texture atlased without one needing to HScript, for the purpose of convenience and reskins.

**Comment by biomseed** [comment]
> Yeah I just found out now. The only question now is if there would be a way to add in objects that are texture atlased without one needing to HScript, for the purpose of convenience and reskins.

You can already do that. Look in the source code, specifically at the character.hx file, for more info. There's comments explaining what things do. In characters, you can use the rendertype "animateatlas" to use an animate atlas for a character. You can't use both spritesheets and atlases in one character yet though.

**Comment by EliteMasterEric** [comment]
Adobe Animate Texture Atlases use JSON to specify individual symbols in an animation, along with an Animation.json to dictate how those symbols move. This is as opposed to the typical Sparrow sprite sheets, which use XML to specify entire frames of animation.

Texture Atlases are much more efficient than Sparrow sheets when symbols are heavily reused, but cannot be generated by anything other than Adobe Animate and don't accurately display certain filters and blend modes when imported in game.

You can instantiate an FlxAtlasSprite to play an Animate Atlas, or use AnimateAtlasCharacter to render a character in a song (see the Blazin' character data files for an example of how to do this). Future functionality will make it easier to mix and match sprite sheets and Atlases, and to use Atlases in stages.

**Comment by Johferson** [comment]
> Adobe Animate Texture Atlases use JSON to specify individual symbols in an animation, along with an Animation.json to dictate how those symbols move. This is as opposed to the typical Sparrow sprite sheets, which use XML to specify entire frames of animation.
> 
> Texture Atlases are much more efficient than Sparrow sheets when symbols are heavily reused, but cannot be generated by anything other than Adobe Animate and don't accurately display certain filters and blend modes when imported in game.
> 
> You can instantiate an FlxAtlasSprite to play an Animate Atlas, or use AnimateAtlasCharacter to render a character in a song (see the Blazin' character data files for an example of how to do this). Future functionality will make it easier to mix and match sprite sheets and Atlases, and to use Atlases in stages.

Thank you so much Mr. Eric, especially on how to do it. I can finally replace my Freeplay BF properly
